core/src/replay_stage.rs

```diff
@@ -2425,7 +2426,15 @@ impl ReplayStage {
 
             let bank_slot = replay_result.bank_slot;
             let bank = &bank_forks.read().unwrap().get(bank_slot).unwrap();
-            if let Some(replay_result) = &replay_result.replay_result {
+            if let Some(mut replay_result) = replay_result.replay_result {
+                // check accounts data size and error if the limits were exceeded
+                // - only check when the bank is complate (for correctness)
+                // - only check when the replay result is OK (for optimization)
+                if bank.is_complete() && replay_result.is_ok() {
+                    if let Err(err) = check_accounts_data_size(bank) {
```

Comment:
This is probably fine from a performance perspective because we call `process_replay_results` infrequent enough (maybe a handful of times per slot).

However, it could be more optimal to move the call to `check_accounts_data_size` inside of the `if bank.is_complete()` conditional below. Advantages would be:

1. Skip this extra check `if bank.is_complete() && replay_result.is_ok()` because we already check them through the match and the existing bank complete checks
2. Don't need to modify the replay_result variable, which means we can pass the replay result vector by reference.

Downside is we would have to duplicate the Err branch handling of `match replay_result` (marking dead slot and continuing)

I'm cool with whatever you decide